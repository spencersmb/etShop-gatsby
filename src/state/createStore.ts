import { IState } from '@et/types/State'
import { breakPointReducer } from '@redux/reducers/breakpointReducer'
import { modalReducer } from '@redux/reducers/modalReducer'
import { productReducer } from '@redux/reducers/productReducer'
import { createStore, applyMiddleware, combineReducers } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunkMiddleware from 'redux-thunk'
import initState from '@redux/reducers/initialState'

const env = process.env.NODE_ENV || 'development'
const initStore = (initialState: IState = initState) => {

	const reducers = combineReducers<IState>({
		breakpoint: breakPointReducer,
		modal: modalReducer,
		products: productReducer,
	})

	if (typeof window !== 'undefined' && env === 'development') {
		return createStore(
			reducers,
			initialState,
			composeWithDevTools(applyMiddleware(thunkMiddleware)),
		)
	} else {
		return createStore(
			reducers,
			initialState,
			composeWithDevTools(applyMiddleware(thunkMiddleware)),

			// applyMiddleware(thunkMiddleware),
		)
	}
}
export default initStore