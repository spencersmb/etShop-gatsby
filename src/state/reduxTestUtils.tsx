// this is a handy function that I normally make available for all my tests
import { ICartState } from '@et/types/Cart'
import { IGatsbyConfig } from '@et/types/Gatsby'
import { IProduct } from '@et/types/Products'
import React from 'react'
import initialState from '@redux/reducers/initialState'
import { Provider } from 'react-redux'
import { render } from 'react-testing-library'
// that deal with connected components.
// you can provide initialState or the entire store that the ui is rendered with
import { createStore } from 'redux'

export function renderWithRedux (
	ui: any,
	reducer: any,
	{ state, store = createStore(reducer, state) }: any = initialState
) {
	return {
		...render(<Provider store={store}>{ui}</Provider>),
		// adding `store` to the returned utilities to allow us
		// to reference it in our tests (just try to avoid using
		// this to test implementation details).
		store
	}
}

const testSiteMetaDataQuery = {
	title: `Every-Tuesday Digital Products Shop`,
	description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
	author: `@Teelac`,
	authorUrl: 'https://every-tuesday.com/about/#teela',
	siteUrl: `https://shop.every-tuesday.com`,
	siteName: `Every-Tuesday Shop`,
	db: 'http://shopeverytuesday.local',
	route: 'et-shop',
	twitterUrl: 'https://twitter.com/teelacunningham',
	twitterDefaultImage: 'url'
}

export const singleItemQuery: IGatsbyConfig & { wcProduct: IProduct } = {
	site: {
		siteMetadata: {
			...testSiteMetaDataQuery
		}
	},
	wcProduct: {
		type: 'simple',
		date_created_gmt: '2018-09-25T20:35:51',
		date_modified_gmt: '2019-02-18T21:44:25',
		id: '202eca74-fc90-56e7-8269-b59f18a19194',
		name: 'Watercolor texture kit Vol. 1',
		price: '16',
		product_id: 222,
		pwyw: false,
		regular_price: '16',
		sale_price: '',
		slug: 'watercolor-texture-kit-vol-1',
		description: 'description',
		short_description: 'short desc',
		on_sale: false,
		features: {
			description: 'feature desc',
			items: {
				description: 'item desc',
				icon: 'icon',
				title: 'singleItemQuery title'
			}
		},
		categories: [
			{ id: 1, slug: 'fonts', name: 'Fonts' },
			{ id: 2, slug: 'Watercolor', name: 'Watercolor' }
		],
		images: [
			{
				id: 123414,
				alt: 'alt',
				fullSize: {
					url: 'fullsize-url'
				},
				thumbnail: {
					url: 'image-url'
				}
			}
		],
		license: {
			hasExtendedLicense: true,
			type: 'standard',
			extendedItem: {
				slug: 'watercolor-texture-kit-vol-1-ext'
			}
		},
		seo: {
			desc: 'Add serious vibrant color and detail with this kit packed with 32 unique watercolor textures, 12 seamless, repeatable watercolor texture patterns and 2 bonus watercolor paper patterns.',
			title: 'Watercolor texture kit Vol. 1'
		},
		tags: [
			{ id: 1, name: 'Texture Kit', slug: 'texture-kit' },
			{ id: 2, name: 'Waterbrush', slug: 'waterbrush' }
		],
		localFile: {
			name: 'file name',
			id: '1234',
			childImageSharp: {
				fluid: {
					src: 'src'
				},
				fixed: {
					width: '400',
					height: '400',
					src: 'src'
				}
			}
		}
	}
}

export const testCartEmpty: ICartState = {
	couponCode: {
		code: 'string',
		discount: 'string',
		loading: false,
		product_ids: [],
		submitted: false,
		type: 'string',
		valid: false,
	},
	isOpen: false,
	items: {
	},
	loaded: true,
	paymentType: 'Stripe',
	totalItems: 0,
	totalPrice: 0
}
