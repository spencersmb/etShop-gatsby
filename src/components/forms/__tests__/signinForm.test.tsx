import React from 'react'
import { connect } from 'react-redux'
import renderer from 'react-test-renderer'
import { cleanup, fireEvent } from 'react-testing-library'
import SignInForm from '@components/forms/signin'
import { reducer, FormReducer } from 'redux-form'
import { renderWithRedux } from '../../../state/reduxTestUtils'
afterEach(cleanup)

const ConnectedFull = connect((state: FormReducer) =>{

		return {
			...state,
		}
	}
)(SignInForm)

const changeFormSpy = jest.fn()
const props = {
	handleSubmit: jest.fn(),
	closeModal: jest.fn(),
	changeForm: changeFormSpy,
	poseRef: React.createRef(),
	firstRender: true
}

const setup = (): any => {
	const mounted = renderWithRedux(<ConnectedFull {...props}/>, reducer)
	const emailInput = mounted.getByLabelText('Email:')
	const passwordInput = mounted.getByLabelText('Password:')
	const submitBtn = mounted.getByText('Submit')
	return {
		submitBtn,
		passwordInput,
		emailInput,
		...mounted,
	}
}

describe('Login Modal', () => {

	it('Should render email input type', ( ) => {
		const mountedForm = setup()
		expect(mountedForm.emailInput.name).toBe('email')
		expect(mountedForm.emailInput.type).toBe('email')
	})

	it('Should render password input type', ( ) => {
		const mountedForm = setup()
		expect(mountedForm.passwordInput.name).toBe('password')
		expect(mountedForm.passwordInput.type).toBe('password')
	})

	it('should have a submit btn', async () => {
		const mountedForm = setup()
		const btn = mountedForm.submitBtn
		btn.click()
		expect(btn).toBeTruthy()
	})

	it('should have a changeForm btn and call changeForm on click', () => {
		const mountedForm = setup()
		const btn = mountedForm.getByText('Sign Up')
		btn.click()

		expect(props.changeForm).toHaveBeenCalledTimes(1)
		expect(btn).toBeTruthy()
	})
})