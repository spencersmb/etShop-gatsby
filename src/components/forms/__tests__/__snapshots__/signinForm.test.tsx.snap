// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Modal Enzyme Should take snapshot of blank form layout 1`] = `"<div data-testid=\\"signIn-form\\" class=\\"sc-htpNat jRBxPX\\"><form><h3>Sign In</h3><div><div style=\\"position: relative;\\"><input name=\\"email\\" aria-label=\\"Email:\\" placeholder=\\"\\" type=\\"email\\" value=\\"\\"><label class=\\"renderLabel sc-bwzfXH hhloFG\\">Email:</label><div class=\\"renderInputSvg sc-bdVaJa etplbY\\"><test-file-stub></test-file-stub></div><span>Required</span></div></div><div><div style=\\"position: relative;\\"><input name=\\"password\\" aria-label=\\"Password:\\" placeholder=\\"\\" type=\\"password\\" value=\\"\\"><label class=\\"renderLabel sc-bwzfXH hhloFG\\">Password:</label><div class=\\"renderInputSvg sc-bdVaJa etplbY\\"><test-file-stub></test-file-stub></div><span>Required</span></div></div><button type=\\"submit\\" disabled=\\"\\">Submit</button></form><button type=\\"button\\" data-form=\\"signup\\">Sign Up</button></div>"`;
